pkg/cmd/release-payload-controller/job_state_controller.go

```diff
@@ -0,0 +1,215 @@
+}
+
+// interrogateJobResults organizes JobRunResults into categorized slices for further processing.  The individual jobs
+// are sorted into the following categories: Pending, Successful, Failed, and Unknown.
+func interrogateJobResults(results []v1alpha1.JobRunResult) (pendingJobs, successfulJobs, failedJobs, unknownJobs []v1alpha1.JobRunResult) {
+	for _, result := range results {
+		switch result.State {
+		case v1alpha1.JobRunStatePending, v1alpha1.JobRunStateTriggered:
+			pendingJobs = append(pendingJobs, result)
+		case v1alpha1.JobRunStateSuccess:
+			successfulJobs = append(successfulJobs, result)
+		case v1alpha1.JobRunStateFailure, v1alpha1.JobRunStateAborted, v1alpha1.JobRunStateError:
+			failedJobs = append(failedJobs, result)
+		default:
+			unknownJobs = append(unknownJobs, result)
+		}
+	}
+
+	return pendingJobs, successfulJobs, failedJobs, unknownJobs
+}
+
+// processJobResults processes JobRunResults for jobs that are not configured as analysis jobs or
+// that have retries.  Any Successful result of this class of jobs returns a "Success" JobState.
+func processJobResults(results []v1alpha1.JobRunResult) v1alpha1.JobState {
+	var pendingJobs, successfulJobs, failedJobs, _ = interrogateJobResults(results)
+
+	switch {
+	case len(successfulJobs) > 0:
+		return v1alpha1.JobStateSuccess
+	case len(pendingJobs) > 0:
+		return v1alpha1.JobStatePending
+	case len(failedJobs) > 0:
+		return v1alpha1.JobStateFailure
+	default:
+		return v1alpha1.JobStateUnknown
+	}
+}
+
+// processAnalysisJobResults processes JobRunResults for analysis jobs.  The JobState will be
+// "Unknown" until the number of JobRunResults equals count.  While any of the JobRunResults are
+// still running (pending), the JobState will be "Pending".  Once all the JobRunResults are complete,
+// a success rate, that is greater-than or equal to analysisSuccessThreshold, will return a "Success"
+// JobState.  Anything less-than analysisSuccessThreshold will return a "Failure" JobState.
+func processAnalysisJobResults(results []v1alpha1.JobRunResult, count int) v1alpha1.JobState {
+	var pendingJobs, successfulJobs, _, _ = interrogateJobResults(results)
+
+	switch {
+	case len(results) != count:
+		return v1alpha1.JobStateUnknown
+	case len(pendingJobs) > 0:
+		return v1alpha1.JobStatePending
+	case (float64(len(successfulJobs)) / float64(count)) >= analysisSuccessThreshold:
+		return v1alpha1.JobStateSuccess
+	default:
+		return v1alpha1.JobStateFailure
+	}
+}
+
+// processRetryJobResults processes JobRunResults for jobs that can be retried.  Any successful
+// JobRunResult will return a "Success" JobStatus.  As long as the number of attempts is less than
+// count and each attempt results in a failure, the JobStatus will return "Pending".  If all retry
+// attempts have failed, the JobStatus will be "Failed".
+func processRetryJobResults(results []v1alpha1.JobRunResult, count int) v1alpha1.JobState {
+	attempts := len(results)
+
+	state := processJobResults(results)
```

Comment:
please rewrite with interrogating here and write the required logic for this method.